components/Post.php

```diff
@@ -113,10 +117,17 @@ protected function getPostSibling($direction = 1)
 
         $postPage = $this->getPage()->getBaseFileName();
 
-        $post->setUrl($postPage, $this->controller);
+        $blogPostComponent = $this->getComponent('blogPost', $postPage);
+        $blogPostsComponent = $this->getComponent('blogPosts', $this->categoryPage);
+
+        $post->setUrl($postPage, $this->controller, [
+            'slug' => $this->urlProperty($blogPostComponent, 'slug')
+        ]);
 
-        $post->categories->each(function($category) {
-            $category->setUrl($this->categoryPage, $this->controller);
+        $post->categories->each(function($category) use ($blogPostsComponent) {
```

Comment:
@GinoPane space between function and arguments as per PSR-2 guidelines.
```suggestion
        $post->categories->each(function ($category) use ($blogPostsComponent) {
```